src/index/Index.cpp

```diff
@@ -84,19 +85,48 @@ void Index::createFromTsvFile(const string& tsvFile, const string& onDiskBase,
   openFileHandles();
 }
 
+// _____________________________________________________________________________________________
+Index::ExtVec Index::createExtVecAndVocabFromNTriples(const string& ntFile, 
+    					       const string& onDiskBase,
+					       bool onDiskLiterals, bool keepTempFiles) {
+  size_t nofLines = passNTriplesFileForVocabulary(ntFile, onDiskLiterals, NUM_TRIPLES_PER_PARTIAL_VOCAB);
+  if (onDiskLiterals) {
+    _vocab.externalizeLiteralsFromTextFile(onDiskBase + EXTERNAL_LITS_TEXT_FILE_NAME, 
+	                                   onDiskBase + ".literals-index");
+  }
+  // clear vocabulary to save ram (only information from partial binary files used from now on).
+  _vocab = Vocabulary();
+  ExtVec v(nofLines);
+  passNTriplesFileIntoIdVector(ntFile, v, onDiskLiterals, NUM_TRIPLES_PER_PARTIAL_VOCAB);
+
+  if (!keepTempFiles) {
+    // remove temporary files only used during index creation
+    LOG(INFO) << "Removing temporary files (partial vocabulary and external text file...\n";
+    string removeCommand1 = "rm " + onDiskBase + EXTERNAL_LITS_TEXT_FILE_NAME;
```

Comment:
use `"rm --" ` to make sure no one can pass e.g. `-rf` as part of `onDiskBase`. Also I'm not sure if we really should use the `system()` function here, sadly the only nice [standard way](https://en.cppreference.com/w/cpp/filesystem/remove) needs C++17. Now that Ubuntu 18.04 is out I'd be willing to use that in QLever though for a lot of our machines that means it really only works in a docker container at the moment. So maybe for now just add a TODO